### PR TITLE
`EncapsulationHeader::to_encoding_i` error messages lack detail

### PR DESCRIPTION
**Pull request by jrw972**
Problem
-------

The error messages from `EncapsulationHeader::to_encoding_i` lack enough detail to actually debug the error.  Typically, the issue is a mismatch in encoding, however, it is unclear which topic, writer, reader, etc. is involved.

Solution
--------

Add the writer and reader guids to the error messages.

### CODE REVIEW

**Review by mitza-oci — CHANGES_REQUESTED** [review]
Instead of moving additional details into the EncapsulationHeader I think we should move to_encoding out to another class or function, maybe in XTypes dir/namespace.

**Comment by iguessthislldo** [comment]
I'm not sure sure about that. `EncapsulationHeader`, like `Encoding` and `Serializer` is aware of XTypes, but isn't specific to XTypes.

**Comment by mitza-oci** [comment]
> I'm not sure sure about that. `EncapsulationHeader`, like `Encoding` and `Serializer` is aware of XTypes, but isn't specific to XTypes.

I was referring to moving the `to_encoding` algorithm, not the `EncapsulationHeader`.  It doesn't have to move to XTypes, but it also doesn't have to be in `Serializer.h`.

**Review by iguessthislldo — APPROVED** [review]

File: dds/DCPS/RTPS/Spdp.cpp [hunk 1]
```diff
@@ -3132,11 +3132,13 @@ Spdp::SpdpTransport::handle_input(ACE_HANDLE h)
         if (data.smHeader.flags & (FLAG_D | FLAG_K_IN_DATA)) {
           DCPS::EncapsulationHeader encap;
           DCPS::Encoding enc;
-          if (!(ser >> encap) || !encap.to_encoding(enc, DCPS::MUTABLE) || enc.kind() != Encoding::KIND_XCDR1) {
+          if (!(ser >> encap) || !to_encoding(enc, encap, DCPS::MUTABLE)  || enc.kind() != Encoding::KIND_XCDR1) {
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/Spdp.cpp:3135]:
```suggestion
          if (!(ser >> encap) || !to_encoding(enc, encap, DCPS::MUTABLE) || enc.kind() != Encoding::KIND_XCDR1) {
```

**Review by mitza-oci — CHANGES_REQUESTED** [review]

**Review by mitza-oci — APPROVED** [review]

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**